pkg/internal/approver/validation/validator.go

```diff
@@ -0,0 +1,87 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/ext"
+)
+
+const (
+	varSelf    = "self"
+	varRequest = "cr"
+)
+
+// Validator knows how to validate CSR attribute values in CertificateRequests
+// against CEL expressions declared in CertificateRequestPolicy.
+// A Validator must be compiled before it can validate. A compiled Validator
+// is stateless, thread-safe, and cacheable.
+type Validator interface {
+	// Compile compiles the Validator CEL expression into a runnable program
+	// that can perform validations. Any compilation errors will be returned.
+	Compile() error
+
+	// Validate validates the supplied value against the Validator CEL
+	// expression in context of the request.
+	// Returns 'true' if the value is valid (passes validation).
+	// A returned error should be considered as an internal/technical error,
+	// and not typically user-facing.
```

Comment:
I presume the goal here is to say "don't show this error to the user, it isn't human-readable", but I don't think that this prose gets you there.

Reply:
You got it. I will try to rephrase this.

Reply:
I have now made an attempt to rewrite this. PTAL.